File: task_equivalents/python/data_manipulation_preprocessing/01_impute_encode_scale.py
# Extracted from Ch09.py (Q7 preprocessing/scaling)

import numpy as np
import pandas as pd
from sklearn.preprocessing import StandardScaler

Auto = pd.read_csv('data/csv/Auto.csv', na_values='?').dropna()

Auto['mpg01'] = np.where(Auto['mpg'] > np.median(Auto['mpg']), 1, 0)
Auto = Auto.drop(['mpg', 'name'], axis=1)

var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']

scaled_var = StandardScaler().fit_transform(Auto[var_to_scale])
df = pd.concat([
    pd.DataFrame(scaled_var, columns=var_to_scale),
    Auto[var_no_scale].reset_index(drop=True)
], axis=1)

x = df.iloc[:, :-1]
y = df['mpg01']

File: task_equivalents/r/data_manipulation_preprocessing/01_impute_encode_scale.R
# Extracted from Ch09.R (Q7 preprocessing/scaling)

Auto = read.csv('data/csv/Auto.csv', header=TRUE, na.strings='?', row.names=1)
Auto = na.omit(Auto)

Auto$mpg01 = ifelse(Auto$mpg > median(Auto$mpg), 1, 0)
Auto = subset(Auto, select=-c(mpg, name))

# Scale continuous variables; keep binary/categorical-coded columns unscaled.
var_no_scale = c('cylinders', 'year', 'origin', 'mpg01')
var_to_scale = c('displacement', 'horsepower', 'weight', 'acceleration')
scaled_var = scale(Auto[, var_to_scale])
df = cbind(as.data.frame(scaled_var), Auto[, var_no_scale])

x = df[, -ncol(df)]
y = df$mpg01
